### PR TITLE
feat(ui): improve responsiveness of KVM VM dropdowns

### LINKED ISSUE #1590: Improve responsiveness of KVMDetails machine list dropdowns
Improvements:
- [x] Dropdown position. It would be nice if ContextualMenu could be given `position="auto"` where the dropdown position would depend on the horizontal position of the toggle.
- [x] Responsive dropdown size. The overall width of the dropdown should look good on all screen sizes.
- [x] Responsive column widths. If we can pass a className to MachineListTable we should be able to give it custom column widths.

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added "auto" position prop to ContextualMenu.
- Gave class names to MachineList cells.
- Added Power col to VMs dropdown.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to KVM details and check that the VMs dropdown shows to the left or right of the toggle depending on where it sits in the viewport.

## Fixes

Fixes #1590

### CODE REVIEW

File: ui/src/app/base/components/ContextualMenu/ContextualMenu.test.js [hunk 1]
```diff
@@ -187,4 +191,27 @@ describe("ContextualMenu ", () => {
...
   });
+
+  it("can dynamically calculate position, defaulting to right", () => {
+    expect(getDropdownPosition()).toBe("right");
+
+    global.innerWidth = 1000;
+    const leftWrapper = {
+      current: {
+        // Middle of toggle is to the left of the viewport center
+        // 300 + 100 / 2 = 400 - less than 500
```
Comment (anthonydillon) [inline ui/src/app/base/components/ContextualMenu/ContextualMenu.test.js:202]: Should be `300 + 200 / 2 = 400`

File: ui/src/app/kvm/components/KVMResourcesCard/_index.scss [hunk 2]
```diff
@@ -5,10 +5,34 @@
   .kvm-machine-list-modal {
     max-width: none;
     min-width: 0;
-    width: 35rem;
+    width: calc(100vw - #{$sp-unit * 10});
```
Comment (anthonydillon) [inline ui/src/app/kvm/components/KVMResourcesCard/_index.scss:8]: Why offset by 80?

File: ui/src/app/machines/views/MachineList/_index.scss [hunk 3]
```diff
@@ -59,52 +59,49 @@
...
-    // Disks
-    &:nth-child(10) {
-      @include breakpoint-widths(0, 0, 8%, 6%, 6%);
-    }
-    // Storage
-    &:nth-child(11) {
-      @include breakpoint-widths(0, 0, 9%, 8%, 6%);
-    }
+  // FQDN/MAC
+  .fqdn-col {
```
Comment (anthonydillon) [inline ui/src/app/machines/views/MachineList/_index.scss:63]: With the new class names I don't think you need the comments above.

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few comments in the code. Will QA shortly.

File: ui/src/app/base/components/ContextualMenu/ContextualMenu.test.js (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/base/components/ContextualMenu/ContextualMenu.test.js:202]: Good catch!

File: ui/src/app/kvm/components/KVMResourcesCard/_index.scss (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/kvm/components/KVMResourcesCard/_index.scss:8]: This was laziness hahaha. The offset is the sum of the gutters on the left and right on x-small screens + the inner horizontal padding of the card. That way the dropdown centers nicely in the card on small screens. I can update with vanilla variables

File: ui/src/app/machines/views/MachineList/_index.scss (hunk 3 above)
Comment (Caleb-Ellis) [reply ui/src/app/machines/views/MachineList/_index.scss:63]: true

**Review by anthonydillon — APPROVED** [review]
Thanks works well :+1: will leave unmerged for you in case there are overlaps with Huws stuff.

**Pull request merged**